storage_controller/src/scheduler.rs

```diff
@@ -102,6 +155,9 @@ impl NodeSchedulingScore for NodeAttachmentSchedulingScore {
                 .get(node_id)
                 .copied()
                 .unwrap_or(AffinityScore::FREE),
+            az_match: preferred_az
```

Comment:
To simplify the Score types, we could transform a None preferred_az to just "true" for az_match (i.e. treat all nodes as if they are in the preferred AZ).  I think eventually the AZ stuff will no longer be optional, right?

Reply:
https://github.com/neondatabase/neon/pull/9083/commits/8d51003b88342dde761fdbaea06e481071e53f63